File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
#include <string>
#include <unordered_set>

class Solution {
public:
    string reverseVowels(string s) {
        int n = s.size();
        int i = 0;
        int j = n-1;
        unordered_set<char> vowels = {
            'a', 'e', 'i', 'o', 'u',
            'A', 'E', 'I', 'O', 'U'
        };
        while (j > i){
            if (vowels.count(s[i])==0){
                cout<<"TRUE, "<<s[i]<<endl;
                i ++;
            }
            else if (vowels.count(s[j])==0){
                cout<<"FALSE, "<<s[j]<<endl;
                j --;
            }
            else{
                swap(s[i], s[j]);
                i++;
                j--;
            }
        }
        return s;
        }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        i = 0
        j = len(s)-1
        vowels = ('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U')
        lists = list(s)
        while j > i:
            if lists[i] not in vowels:
                i+=1
            elif lists[j] not in vowels:
                j-=1
            else:
                lists[i], lists[j] = lists[j], lists[i]
                i += 1
                j -= 1
        return ''.join(lists)